app.js

```diff
@@ -112,6 +113,21 @@ let App = create({
         </div>
         <TestResults json={this.state.json}/>
       </>
+
+      // Display server version information
+      const serverVersion = this.state.json.Version;
+      let versionText;
+      if (serverVersion == undefined) {
+        versionText = "This homeserver does not supply version information"
+      } else if (serverVersion.error != undefined) {
+        versionText = serverVersion.error;
+      } else {
+        versionText = serverVersion.name + " " + serverVersion.version;
+      }
+      
+      version = <>
+        <p>Version information: {versionText}</p>
```

Comment:
```suggestion
        <p>Homeserver version: {versionText}</p>
```

Might be better to keep that part specific to the "we got a version and no error" case in the `if...else if...else` blocks above.